### DISCUSSION TITLE
How to get actual Uri if receiving NotFound

### DISCUSSION THREAD

**Discussion by KP-WhatEver** [OrchardCMS/OrchardCore#12416] [Q&A] [answered]
I would like to log the requested uri that could not be found.  
But when accessing the context in a NotFound.cshtml or implementation of IAsyncAlwaysRunResultFilter, the uri only points to /Error/x.

Can someone tell me how to access the actually requested uri, please? :)

**Comment by hishamco** [comment]
Perhaps we could log into diagnostics module, but you could write a middleware that checks the response code status 404 then you can log the requested URL

**Reply by KP-WhatEver** [reply]
Thank you for your answer and for pointing me to towards middleware!
I have the same problem there; the middleware is first called with status code 200 and the correct path, but then with status code 404 and the path is again set to "/Error/404".
Is there something I overlooked?

**Comment by jtkech** [comment, marked as answer]
We have an HttpFeature holding the OriginalPathBase and OriginalPath that you can retrieve from the current HttpContext, e.g. from a razor view.

@using OrchardCore.Environment.Shell

    var shellFeature = Context.Features.Get<ShellContextFeature>();

    var pathBase = shellFeature.OriginalPathBase;
    var path = shellFeature.OriginalPath;
    var fullPath = pathBase + path;

**Reply by KP-WhatEver** [reply]
That's perfect, thank you! :)
